### PR TITLE
Remove `--repo-name` argument for service pipeline installations

### LINKED ISSUE #934: Need better error messages for install-lifecycle-pipeline
I attempted to create an install-lifecycle-pipeline with spk and was greeted with an error message that doesn't give me a lot to go on in terms of how to fix it:

$ spk project install-lifecycle-pipeline --org-name tpark --devops-project spktest --repo-url https://github.com/timfpark/testService --repo-name "test-service-repo" --pipeline-name test-service-lifecycle-pipeline --personal-access-token mh3jiepsfgfq7deadbeefuo57itpeupwxw7cegucrlrrj7xsba
(node:29134) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
info:    Fetched DevOps Client
info:    Creating pipeline for definition
error:   Error creating BuildDefinition; buildApi.createDefinition() returned an invalid value of null
error:   Error occurred during pipeline creation for test-service-lifecycle-pipeline
error:   Error creating definition

Can we add parameter validation and/or better troubleshooting metadata to improve on this?

### PR DESCRIPTION
**Pull request by yradsmikham**
Close https://github.com/microsoft/bedrock/issues/934.

### CODE REVIEW

**Review by bnookala — CHANGES_REQUESTED** [review]
I think you may need to modify the integration tests as well, as this (removing an argument) will likely cause it to fail

**Review by mtarng — APPROVED** [review]

File: src/lib/gitutils.ts [hunk 1]
```diff
@@ -301,3 +301,17 @@ export const checkoutCommitPushCreatePRLink = async (
...
 };
+
+/**
+ * Validates whether a url is a github url
+ * TEMPORARY, UNTIL GITHUB REPOS ARE SUPPORTED
+ *
+ * @param url the url string
+ */
+export const isGitHubUrl = (url: string) => {
+  const gitUrl = url.includes("github");
```
Comment (bnookala) [inline src/lib/gitutils.ts]: I think we should use a url parser here instead of string matching ie https://nodejs.org/api/url.html#url_url_hostname

**Review by bnookala — CHANGES_REQUESTED** [review]
small thing

File: src/lib/gitutils.ts (hunk 1 above)
Comment (dennisseah) [reply src/lib/gitutils.ts]: @bnookala , we have followed your suggestion on parsing the url string and check for host name.

**Review by dennisseah — APPROVED** [review]

**Review by NathanielRose — APPROVED** [review]

**Review by bnookala — APPROVED** [review]

**Pull request merged**